0.43

balloob opened the pull request:
Oooh yeah. It's time for 0.43 and this is going to be a killer release. For the people that have been following on social media, you might have noticed that [we got pretty excited about the new IKEA Tradfri line up][ikea-blog]. And so we are very happy to announce that thanks to the work by [@ggravlingen], [@MartinHjelmare] and myself that this will be the first release to support the IKEA light bulbs. Home Assistant will automatically detect the gateway on your network and after following the instructions, allow you to control your lights.

![discovery-1](https://cloud.githubusercontent.com/assets/1444314/25301319/7a21bb16-26d8-11e7-9d91-e52445cc715a.png)

And in case you have missed the other big news: [@robbiet480] released the [first version of our iOS app][ios]! It took a little over a year but it offers some great ways to make your iDevice work with Home Assistant. Big congrats to Robbie for shipping! As with our other parts, the app is open-sourced under APACHE 2 and contributors are welcome. [Check it out.][ios-source]

A big shout out this release to [@happyleavesaoc] for his constant stream of great contributions to Home Assistant. The first contribution was back in October 2015 and since then @happyleavesaoc has been responsible for many platforms and bug fixes including the Spotify platform this release. Thanks @happyleavesaoc, you're a great example of what makes our community awesome.

The new Spotify integration is based on the new Spotify Connect Playback API. It supports playback, album art and switching which Spotify device you're controlling.

[ikea-blog]: https://home-assistant.io/blog/2017/04/17/ikea-tradfri-internet-of-things-done-right/
[ios]: https://home-assistant.io/blog/2017/04/15/ios/
[ios-source]: https://github.com/home-assistant/home-assistant-ios

## New Platforms

- Add MaryTTS platform ([@johanpalmqvist] - [#6988]) ([tts.marytts docs]) (new-platform)
- Telegram bot component (incl. webhook and polling platform) ([@sander76] - [#6913]) ([telegram_bot docs]) ([telegram_bot.polling docs]) ([telegram_bot.webhooks docs]) (breaking change) (new-platform)
- Add support fo map data from Neato ([@turbokongen] - [#6939]) ([neato docs]) ([camera.neato docs]) ([sensor.neato docs]) (new-platform)
- Added initial support for IKEA Tradfri Gateway ([@ggravlingen] - [#7074]) ([light.tradfri docs]) (new-platform)
- MQTT camera ([@MrMep] - [#7092]) ([camera.mqtt docs]) (new-platform)
- Add LIFX Cloud scene support ([@amelchio] - [#7124]) ([scene.lifx_cloud docs]) (new-platform)
- spotify media player ([@happyleavesaoc] - [#6980]) ([media_player.spotify docs]) (new-platform)
- JSON MQTT Device tracker ([@MrMep] - [#7055]) (new-platform)
- opensky sensor ([@happyleavesaoc] - [#7061]) ([sensor.opensky docs]) (new-platform)
- Add ping binary sensor ([@fabaff] - [#7052]) ([binary_sensor.ping docs]) (new-platform)

## Breaking changes

- Remove deprecated classes from `homeassistant.remote` ([@balloob] - [#7011])
- min_max sensor has a different default name and will ignore states that are unknown ([@micw] - [#6786])
- `telegram_webhooks` has been converted to a new `telegram_bot` component. It has two platforms: webhook and polling platform ([@sander76] - [#6913]) ([telegram_bot docs])([telegram_bot.polling docs]) ([telegram_bot.webhooks docs])

```yaml
telegram_bot:
    platform: webhooks
    api_key : api_key_here
    allowed_chat_ids : 
        - 123456
        - 456789
```

- Lutron_Caseta: it is no longer necessary to specify username and password in the config. ([@gurumitts] - [#7165]) ([lutron_caseta docs]) ([light.lutron_caseta docs])
- mvglive: config has been updated to allow support for multiple departures ([@mountainsandcode] - [#6953]) ([sensor.mvglive docs])

```yaml
sensor:
  - platform: mvglive
    nextdeparture:
     - station: Hauptbahnhof
       name: Hbf
       destinations: ['München Flughafen Terminal','Markt Schwaben']
       products: 'S-Bahn'
       timeoffset: 2
     - station: Sendlinger Tor
       lines: ['U2','U8']
     - station: Scheidplatz
       products: ['U-Bahn']
       directions: '1'
```

## All changes

- Update kodi for aiohttp2 ([@armills] - [#6967]) ([media_player.kodi docs]) ([notify.kodi docs])
- Fix current_temperature is rounded ([@aufano] - [#6960]) ([climate docs])
- Bugfix time and task coro ([@pvizeli] - [#6968])
- Initial import for HassIO ([@pvizeli] - [#6935]) ([hassio docs]) (new-platform)
- Preserve customize glob order. ([@andrey-git] - [#6963])
- Foscam Camera: Adding exception handling when fetching the camera image to avoid python exception errors when host is not reachable or rather any url error to camera ([@viswa-swami] - [#6964]) ([camera.foscam docs])
- light.yeelight: catch i/o related exceptions from the backend lib ([@rytilahti] - [#6952])
- From Dusk till Dawn ([@BillyNate] - [#6857]) ([sun docs])
- Tests for zwave services ([@armills] - [#6937]) ([zwave docs])
- Fix control+c quitting HASS ([@balloob] - [#6974])
- Update Emby for aiohttp v2 ([@mezz64] - [#6981])
- switch.tplink: upgrade to the newest upstream release which adds support for plugs using the newer communication protocol ([@rytilahti] - [#6979]) ([switch.tplink docs])
- switch.tplink: bump pyhs100 version requirement ([@rytilahti] - [#6986])
- Add tests for ZWaveDeviceEntityValues helper ([@armills] - [#6978]) ([zwave docs])
- Bump Amcrest module to 1.1.8 ([@tchellomello] - [#6990])
- update gstreamer ([@happyleavesaoc] - [#6987]) ([media_player.gstreamer docs])
- Warn if start takes a long time. ([@balloob] - [#6975])
- Upgrade to aiohttp 2.0.6 ([@balloob] - [#6992])
- Make discovery not block start ([@balloob] - [#6991])
- Downgrade aiohttp to 205 ([@balloob] - [#6994])
- Bump pywemo version. ([@pavoni] - [#7004])
- Fix Synology camera content type ([@balloob] - [#7010]) ([camera.synology docs])
- Fix two more instances of JSON parsing synology ([@balloob] - [#7014])
- Bump pyalarmdotcom to support new version of aiohttp ([@Xorso] - [#7021]) ([alarm_control_panel.alarmdotcom docs])
- Fix US states check (fixes #7015) ([@fabaff] - [#7017])
- Remove deprecated remote classes ([@balloob] - [#7011]) (breaking change)
- Replace 'vendor_id' with 'arch' (fixes #7003) ([@fabaff] - [#7023])
- more tests for slugify ([@micw] - [#7027])
- Additional ZWave coverage ([@armills] - [#7024]) ([zwave docs])
- bump ups version ([@happyleavesaoc] - [#7033]) ([sensor.ups docs])
- update usps version ([@happyleavesaoc] - [#7035]) ([sensor.usps docs])
- update fedex ([@happyleavesaoc] - [#7034]) ([sensor.fedex docs])
- Google TTS can't read percent sign (#6971) ([@pezinek] - [#7030]) ([tts.google docs])
- Feature/min max improvements ([@micw] - [#6786]) (breaking change)
- Upgrade psutil to 5.2.2 ([@fabaff] - [#7037]) ([sensor.systemmonitor docs])
- Upgrade sendgrid to 4.0.0 ([@fabaff] - [#7038]) ([notify.sendgrid docs])
- Missing line name restriction added (fixes #7039) ([@DavidMStraub] - [#7040]) ([sensor.mvglive docs])
- Plug file leak on LIFX unregister ([@amelchio] - [#7031]) ([light.lifx docs])
- Make core to look avilable state of device on servicecall ([@pvizeli] - [#7045])
- Remove configuration sample ([@fabaff] - [#7048])
- Bugfix wait on start event ([@pvizeli] - [#7013])
- Bugfix slider ([@pvizeli] - [#7047]) ([input_slider docs])
- Add MaryTTS platform ([@johanpalmqvist] - [#6988]) ([tts.marytts docs]) (new-platform)
- Fix mysensors callback ([@MartinHjelmare] - [#7057]) ([mysensors docs]) ([device_tracker.mysensors docs])
- Constrain chardet to 2.3 ([@balloob] - [#7063])
- Fix/slugify with german umlaut ss ([@micw] - [#7029])
- Speed up aiohttp ([@balloob] - [#7064])
- Upgrade netdisco to 1.0.0rc2 ([@balloob] - [#7008])
- Telegram bot component (incl. webhook and polling platform) ([@sander76] - [#6913]) ([telegram_bot docs]) ([telegram_bot.polling docs]) ([telegram_bot.webhooks docs]) (breaking change) (new-platform)
- Upgrade paho-mqtt to 1.2.2 ([@fabaff] - [#7066])
- Fix handling with register callbacks on added_to_hass ([@pvizeli] - [#7067])
- Lutron. Bugfix callback registration. ([@pvizeli] - [#7042])
- Adding AlarmDecoder platform ([@hawk259] - [#6900])
- Add communication data attributes to Zwave node_entity ([@turbokongen] - [#6813]) ([zwave docs]) ([zwave.node_entity docs])
- Add product_name attribute to zwave nodes. ([@andrey-git] - [#7071]) ([zwave.node_entity docs])
- Bump braviarc version to 0.3.7 ([@robbiet480] - [#7078]) ([media_player.braviatv docs])
- Fix account balance in fido sensor ([@titilambert] - [#7077]) ([sensor.fido docs])
- MQTT: Managing binary payloads ([@MrMep] - [#6976]) ([mqtt docs])
- Bump qnapstats library version to 0.2.4 ([@colinodell] - [#7085]) ([sensor.qnap docs])
- Add support fo map data from Neato ([@turbokongen] - [#6939]) ([neato docs]) ([camera.neato docs]) ([sensor.neato docs]) (new-platform)
- Added initial support for IKEA Tradfri Gateway ([@ggravlingen] - [#7074]) ([light.tradfri docs]) (new-platform)
- Better thread safety in zwave node_entity ([@andrey-git] - [#7087]) ([zwave.node_entity docs])
- Uber version bump ([@armills] - [#7100]) ([sensor.uber docs])
- No product ids configured should fetch all ids ([@armills] - [#7091])
- Upgrade aiohttp to 2.0.7 ([@fabaff] - [#7106])
- Update file header, add const for defaults, and update log messages ([@fabaff] - [#7110])
- Upgrade speedtest-cli to 1.0.4 ([@fabaff] - [#7105]) ([sensor.speedtest docs])
- Remove globally disabled pylint issue and update docstrings ([@fabaff] - [#7111])
- Use third-party lib aioautomatic for automatic ([@armills] - [#7126]) ([device_tracker.automatic docs])
- Upgrade chardet to 3.0.2 ([@fabaff] - [#7112])
- Send stderr of ping tracker to devnull ([@amelchio] - [#7096]) ([device_tracker.ping docs])
- Upgraded Amcrest module to 1.1.9 to support new firmware versions: ([@tchellomello] - [#7130])
- Mqtt camera ([@MrMep] - [#7092]) ([camera.mqtt docs]) (new-platform)
- Bump version to 0.43.0.dev0 ([@MartinHjelmare] - [#7132])
- Load zwave panel ([@balloob] - [#7127]) ([zwave docs])
- IKEA Tradfri Gateway: added support for RGB ([@ggravlingen] - [#7115]) ([light.tradfri docs])
- Make Tradfri discoverable ([@balloob] - [#7128])
- Make version number optional and a string to fix identify issue introduced in iOS 1.0.1 ([@robbiet480] - [#7141])
- Fix for zwave RGB setting ([@armills] - [#7137])
- Replace rollershutter with cover in demo ([@robbiet480] - [#7140])
- Add debug logging to automation initial state ([@armills] - [#7068]) ([automation docs])
- Always return True/False from is_state and is_state_attr ([@amelchio] - [#7138])
- Add LIFX Cloud scene support ([@amelchio] - [#7124]) ([scene.lifx_cloud docs]) (new-platform)
- Fix LIFX lights with disappearing names ([@amelchio] - [#7119]) ([light.lifx docs])
- lutron: fix typo that prevented callback registration ([@thecynic] - [#7148]) ([lutron docs])
- Fix HassIO timeout bug ([@pvizeli] - [#7155]) ([hassio docs])
- small fix for random effect in order to use the whole rgb range. So 255 is not excluded anymore. ([@Mister-Espria] - [#7156]) ([light.flux_led docs]) ([light.yeelight docs])
- exposed content_type in rest_command ([@cmsimike] - [#7101])
- Add vera power meter. ([@pavoni] - [#7134]) ([vera docs]) ([sensor.vera docs])
- Add support for tradfri color temp ([@MartinHjelmare] - [#7153]) ([light.tradfri docs])
- Disable MQTT camera test ([@balloob] - [#7164])
- myStrom WiFi bulbs ([@fabaff] - [#7161])
- Fix #7026 adding a new wol parameter ([@titilambert] - [#7144]) ([switch.wake_on_lan docs])
- Add subscription update for Wemo switches, fix bug in Insight switches, fix wemo motion bug, fix wemo discovery ([@pavoni] - [#7135]) ([wemo docs])
- Update neato.py ([@michaelarnauts] - [#7166]) ([neato docs])
- Keep track of already added players ([@molobrakos] - [#7149])
- Fix id zone mismatch ([@gurumitts] - [#7165]) ([lutron_caseta docs]) ([light.lutron_caseta docs]) (breaking change)
- Upgrade pytradfri to 1.0 ([@balloob] - [#7163])
- Value of 0 should not be considered `unknown`. ([@aequitas] - [#7139])
- Upgrade netdisco ([@balloob] - [#7171])
- Added new services to platform kodi ([@alexmogavero] - [#6426]) ([media_player docs]) ([media_player.kodi docs])
- Supress trackback and upgrade PyMata to 2.14 ([@fabaff] - [#7176]) ([arduino docs])
- Disable invalid-sequence-index ([@fabaff] - [#7177])
- Upgrade mypy to 0.501 (was renamed from mypy-lang) ([@fabaff] - [#7117])
- Add condition for API failure ([@sytone] - [#7181])
- Add history to component priority list ([@balloob] - [#7173])
- Tweak Tradfri ([@balloob] - [#7172]) ([light.tradfri docs])
- Mqtt camera test ([@MrMep] - [#7175])
- updated pylgtv module to fix problems with timeouts ([@hmn] - [#7184]) ([media_player.webostv docs]) ([notify.webostv docs])
- Fix wemo discovery ([@balloob] - [#7183]) ([wemo docs]) ([binary_sensor.wemo docs]) ([light.wemo docs]) ([switch.wemo docs])
- Add Bose soundtouch discovery support and upgrade libsoundtouch library ([@CharlesBlonde] - [#7005])
- spotify media player ([@happyleavesaoc] - [#6980]) ([media_player.spotify docs]) (new-platform)
- JSON MQTT Device tracker ([@MrMep] - [#7055]) (new-platform)
- opensky sensor ([@happyleavesaoc] - [#7061]) ([sensor.opensky docs]) (new-platform)
- Add ping binary sensor ([@fabaff] - [#7052]) ([binary_sensor.ping docs]) (new-platform)
- Added light.pwm component. ([@soldag] - [#7009])
- Fix for errors on missing preview on LG webos TV ([@masarliev] - [#6755]) ([media_player.webostv docs])
- Fix auto discovery for Apple TV ([@postlund] - [#7188]) ([media_player.apple_tv docs])
- Do not request artwork if not available ([@postlund] - [#7189]) ([media_player.apple_tv docs])
- mvglive bug fixes and improvements ([@mountainsandcode] - [#6953]) ([sensor.mvglive docs]) (breaking change)
- Upgrade py-cpuinfo to 3.2.0 ([@fabaff] - [#7190]) ([sensor.cpuspeed docs])
- Add support of input registers while querying modbus sensor. ([@LvivEchoes] - [#7082]) ([modbus docs]) ([sensor.modbus docs])
- Add HassIO to discovery component ([@pvizeli] - [#7195])
- LIFX light effects ([@amelchio] - [#7145]) ([light.lifx docs]) ([light.services.yaml docs])
- HassIO API v2 ([@pvizeli] - [#7201])
- Support xy_color with LIFX lights ([@amelchio] - [#7208]) ([light.lifx docs])
- Update ios.py ([@biacz] - [#7160])
- Fix arwn platform to update hass state when events are received ([@sdague] - [#7202])
- Issue 6749 updated pylgtv to 0.1.6 to fix thread leak in asyncio loop ([@hmn] - [#7199]) ([media_player.webostv docs]) ([notify.webostv docs])

[#6426]: https://github.com/home-assistant/home-assistant/pull/6426
[#6755]: https://github.com/home-assistant/home-assistant/pull/6755
[#6786]: https://github.com/home-assistant/home-assistant/pull/6786
[#6813]: https://github.com/home-assistant/home-assistant/pull/6813
[#6857]: https://github.com/home-assistant/home-assistant/pull/6857
[#6900]: https://github.com/home-assistant/home-assistant/pull/6900
[#6913]: https://github.com/home-assistant/home-assistant/pull/6913
[#6935]: https://github.com/home-assistant/home-assistant/pull/6935
[#6937]: https://github.com/home-assistant/home-assistant/pull/6937
[#6939]: https://github.com/home-assistant/home-assistant/pull/6939
[#6952]: https://github.com/home-assistant/home-assistant/pull/6952
[#6953]: https://github.com/home-assistant/home-assistant/pull/6953
[#6960]: https://github.com/home-assistant/home-assistant/pull/6960
[#6963]: https://github.com/home-assistant/home-assistant/pull/6963
[#6964]: https://github.com/home-assistant/home-assistant/pull/6964
[#6966]: https://github.com/home-assistant/home-assistant/pull/6966
[#6967]: https://github.com/home-assistant/home-assistant/pull/6967
[#6968]: https://github.com/home-assistant/home-assistant/pull/6968
[#6974]: https://github.com/home-assistant/home-assistant/pull/6974
[#6975]: https://github.com/home-assistant/home-assistant/pull/6975
[#6976]: https://github.com/home-assistant/home-assistant/pull/6976
[#6978]: https://github.com/home-assistant/home-assistant/pull/6978
[#6979]: https://github.com/home-assistant/home-assistant/pull/6979
[#6980]: https://github.com/home-assistant/home-assistant/pull/6980
[#6981]: https://github.com/home-assistant/home-assistant/pull/6981
[#6986]: https://github.com/home-assistant/home-assistant/pull/6986
[#6987]: https://github.com/home-assistant/home-assistant/pull/6987
[#6988]: https://github.com/home-assistant/home-assistant/pull/6988
[#6990]: https://github.com/home-assistant/home-assistant/pull/6990
[#6991]: https://github.com/home-assistant/home-assistant/pull/6991
[#6992]: https://github.com/home-assistant/home-assistant/pull/6992
[#6994]: https://github.com/home-assistant/home-assistant/pull/6994
[#7004]: https://github.com/home-assistant/home-assistant/pull/7004
[#7005]: https://github.com/home-assistant/home-assistant/pull/7005
[#7008]: https://github.com/home-assistant/home-assistant/pull/7008
[#7009]: https://github.com/home-assistant/home-assistant/pull/7009
[#7010]: https://github.com/home-assistant/home-assistant/pull/7010
[#7011]: https://github.com/home-assistant/home-assistant/pull/7011
[#7013]: https://github.com/home-assistant/home-assistant/pull/7013
[#7014]: https://github.com/home-assistant/home-assistant/pull/7014
[#7017]: https://github.com/home-assistant/home-assistant/pull/7017
[#7021]: https://github.com/home-assistant/home-assistant/pull/7021
[#7023]: https://github.com/home-assistant/home-assistant/pull/7023
[#7024]: https://github.com/home-assistant/home-assistant/pull/7024
[#7027]: https://github.com/home-assistant/home-assistant/pull/7027
[#7029]: https://github.com/home-assistant/home-assistant/pull/7029
[#7030]: https://github.com/home-assistant/home-assistant/pull/7030
[#7031]: https://github.com/home-assistant/home-assistant/pull/7031
[#7033]: https://github.com/home-assistant/home-assistant/pull/7033
[#7034]: https://github.com/home-assistant/home-assistant/pull/7034
[#7035]: https://github.com/home-assistant/home-assistant/pull/7035
[#7037]: https://github.com/home-assistant/home-assistant/pull/7037
[#7038]: https://github.com/home-assistant/home-assistant/pull/7038
[#7040]: https://github.com/home-assistant/home-assistant/pull/7040
[#7042]: https://github.com/home-assistant/home-assistant/pull/7042
[#7045]: https://github.com/home-assistant/home-assistant/pull/7045
[#7047]: https://github.com/home-assistant/home-assistant/pull/7047
[#7048]: https://github.com/home-assistant/home-assistant/pull/7048
[#7052]: https://github.com/home-assistant/home-assistant/pull/7052
[#7055]: https://github.com/home-assistant/home-assistant/pull/7055
[#7057]: https://github.com/home-assistant/home-assistant/pull/7057
[#7061]: https://github.com/home-assistant/home-assistant/pull/7061
[#7063]: https://github.com/home-assistant/home-assistant/pull/7063
[#7064]: https://github.com/home-assistant/home-assistant/pull/7064
[#7066]: https://github.com/home-assistant/home-assistant/pull/7066
[#7067]: https://github.com/home-assistant/home-assistant/pull/7067
[#7068]: https://github.com/home-assistant/home-assistant/pull/7068
[#7071]: https://github.com/home-assistant/home-assistant/pull/7071
[#7074]: https://github.com/home-assistant/home-assistant/pull/7074
[#7077]: https://github.com/home-assistant/home-assistant/pull/7077
[#7078]: https://github.com/home-assistant/home-assistant/pull/7078
[#7082]: https://github.com/home-assistant/home-assistant/pull/7082
[#7085]: https://github.com/home-assistant/home-assistant/pull/7085
[#7087]: https://github.com/home-assistant/home-assistant/pull/7087
[#7091]: https://github.com/home-assistant/home-assistant/pull/7091
[#7092]: https://github.com/home-assistant/home-assistant/pull/7092
[#7096]: https://github.com/home-assistant/home-assistant/pull/7096
[#7100]: https://github.com/home-assistant/home-assistant/pull/7100
[#7101]: https://github.com/home-assistant/home-assistant/pull/7101
[#7105]: https://github.com/home-assistant/home-assistant/pull/7105
[#7106]: https://github.com/home-assistant/home-assistant/pull/7106
[#7110]: https://github.com/home-assistant/home-assistant/pull/7110
[#7111]: https://github.com/home-assistant/home-assistant/pull/7111
[#7112]: https://github.com/home-assistant/home-assistant/pull/7112
[#7115]: https://github.com/home-assistant/home-assistant/pull/7115
[#7117]: https://github.com/home-assistant/home-assistant/pull/7117
[#7119]: https://github.com/home-assistant/home-assistant/pull/7119
[#7124]: https://github.com/home-assistant/home-assistant/pull/7124
[#7126]: https://github.com/home-assistant/home-assistant/pull/7126
[#7127]: https://github.com/home-assistant/home-assistant/pull/7127
[#7128]: https://github.com/home-assistant/home-assistant/pull/7128
[#7130]: https://github.com/home-assistant/home-assistant/pull/7130
[#7132]: https://github.com/home-assistant/home-assistant/pull/7132
[#7134]: https://github.com/home-assistant/home-assistant/pull/7134
[#7135]: https://github.com/home-assistant/home-assistant/pull/7135
[#7137]: https://github.com/home-assistant/home-assistant/pull/7137
[#7138]: https://github.com/home-assistant/home-assistant/pull/7138
[#7139]: https://github.com/home-assistant/home-assistant/pull/7139
[#7140]: https://github.com/home-assistant/home-assistant/pull/7140
[#7141]: https://github.com/home-assistant/home-assistant/pull/7141
[#7144]: https://github.com/home-assistant/home-assistant/pull/7144
[#7145]: https://github.com/home-assistant/home-assistant/pull/7145
[#7148]: https://github.com/home-assistant/home-assistant/pull/7148
[#7149]: https://github.com/home-assistant/home-assistant/pull/7149
[#7153]: https://github.com/home-assistant/home-assistant/pull/7153
[#7155]: https://github.com/home-assistant/home-assistant/pull/7155
[#7156]: https://github.com/home-assistant/home-assistant/pull/7156
[#7160]: https://github.com/home-assistant/home-assistant/pull/7160
[#7161]: https://github.com/home-assistant/home-assistant/pull/7161
[#7163]: https://github.com/home-assistant/home-assistant/pull/7163
[#7164]: https://github.com/home-assistant/home-assistant/pull/7164
[#7165]: https://github.com/home-assistant/home-assistant/pull/7165
[#7166]: https://github.com/home-assistant/home-assistant/pull/7166
[#7171]: https://github.com/home-assistant/home-assistant/pull/7171
[#7172]: https://github.com/home-assistant/home-assistant/pull/7172
[#7173]: https://github.com/home-assistant/home-assistant/pull/7173
[#7175]: https://github.com/home-assistant/home-assistant/pull/7175
[#7176]: https://github.com/home-assistant/home-assistant/pull/7176
[#7177]: https://github.com/home-assistant/home-assistant/pull/7177
[#7181]: https://github.com/home-assistant/home-assistant/pull/7181
[#7183]: https://github.com/home-assistant/home-assistant/pull/7183
[#7184]: https://github.com/home-assistant/home-assistant/pull/7184
[#7188]: https://github.com/home-assistant/home-assistant/pull/7188
[#7189]: https://github.com/home-assistant/home-assistant/pull/7189
[#7190]: https://github.com/home-assistant/home-assistant/pull/7190
[#7195]: https://github.com/home-assistant/home-assistant/pull/7195
[#7199]: https://github.com/home-assistant/home-assistant/pull/7199
[#7201]: https://github.com/home-assistant/home-assistant/pull/7201
[#7202]: https://github.com/home-assistant/home-assistant/pull/7202
[#7208]: https://github.com/home-assistant/home-assistant/pull/7208
[@BillyNate]: https://github.com/BillyNate
[@CharlesBlonde]: https://github.com/CharlesBlonde
[@DavidMStraub]: https://github.com/DavidMStraub
[@JesseWebDotCom]: https://github.com/JesseWebDotCom
[@LvivEchoes]: https://github.com/LvivEchoes
[@MartinHjelmare]: https://github.com/MartinHjelmare
[@Mister-Espria]: https://github.com/Mister-Espria
[@MrMep]: https://github.com/MrMep
[@Xorso]: https://github.com/Xorso
[@aequitas]: https://github.com/aequitas
[@alexmogavero]: https://github.com/alexmogavero
[@amelchio]: https://github.com/amelchio
[@andrey-git]: https://github.com/andrey-git
[@armills]: https://github.com/armills
[@aufano]: https://github.com/aufano
[@balloob]: https://github.com/balloob
[@biacz]: https://github.com/biacz
[@cmsimike]: https://github.com/cmsimike
[@colinodell]: https://github.com/colinodell
[@fabaff]: https://github.com/fabaff
[@ggravlingen]: https://github.com/ggravlingen
[@gurumitts]: https://github.com/gurumitts
[@happyleavesaoc]: https://github.com/happyleavesaoc
[@hawk259]: https://github.com/hawk259
[@hmn]: https://github.com/hmn
[@johanpalmqvist]: https://github.com/johanpalmqvist
[@masarliev]: https://github.com/masarliev
[@mezz64]: https://github.com/mezz64
[@michaelarnauts]: https://github.com/michaelarnauts
[@micw]: https://github.com/micw
[@molobrakos]: https://github.com/molobrakos
[@mountainsandcode]: https://github.com/mountainsandcode
[@pavoni]: https://github.com/pavoni
[@pezinek]: https://github.com/pezinek
[@postlund]: https://github.com/postlund
[@pvizeli]: https://github.com/pvizeli
[@robbiet480]: https://github.com/robbiet480
[@rytilahti]: https://github.com/rytilahti
[@sander76]: https://github.com/sander76
[@sdague]: https://github.com/sdague
[@soldag]: https://github.com/soldag
[@sytone]: https://github.com/sytone
[@tchellomello]: https://github.com/tchellomello
[@thecynic]: https://github.com/thecynic
[@titilambert]: https://github.com/titilambert
[@turbokongen]: https://github.com/turbokongen
[@viswa-swami]: https://github.com/viswa-swami
[alarm_control_panel.alarmdotcom docs]: https://home-assistant.io/components/alarm_control_panel.alarmdotcom
[arduino docs]: https://home-assistant.io/components/arduino
[automation docs]: https://home-assistant.io/components/automation
[binary_sensor.ping docs]: https://home-assistant.io/components/binary_sensor.ping
[binary_sensor.wemo docs]: https://home-assistant.io/components/binary_sensor.wemo
[camera.foscam docs]: https://home-assistant.io/components/camera.foscam
[camera.mqtt docs]: https://home-assistant.io/components/camera.mqtt
[camera.neato docs]: https://home-assistant.io/components/camera.neato
[camera.synology docs]: https://home-assistant.io/components/camera.synology
[climate docs]: https://home-assistant.io/components/climate
[device_tracker.automatic docs]: https://home-assistant.io/components/device_tracker.automatic
[device_tracker.mysensors docs]: https://home-assistant.io/components/device_tracker.mysensors
[device_tracker.ping docs]: https://home-assistant.io/components/device_tracker.ping
[hassio docs]: https://home-assistant.io/components/hassio
[input_slider docs]: https://home-assistant.io/components/input_slider
[light.flux_led docs]: https://home-assistant.io/components/light.flux_led
[light.lifx docs]: https://home-assistant.io/components/light.lifx
[light.lutron_caseta docs]: https://home-assistant.io/components/light.lutron_caseta
[light.services.yaml docs]: https://home-assistant.io/components/light.services.yaml
[light.tradfri docs]: https://home-assistant.io/components/light.tradfri
[light.wemo docs]: https://home-assistant.io/components/light.wemo
[light.yeelight docs]: https://home-assistant.io/components/light.yeelight
[lutron docs]: https://home-assistant.io/components/lutron
[lutron_caseta docs]: https://home-assistant.io/components/lutron_caseta
[media_player docs]: https://home-assistant.io/components/media_player
[media_player.apple_tv docs]: https://home-assistant.io/components/media_player.apple_tv
[media_player.braviatv docs]: https://home-assistant.io/components/media_player.braviatv
[media_player.gstreamer docs]: https://home-assistant.io/components/media_player.gstreamer
[media_player.kodi docs]: https://home-assistant.io/components/media_player.kodi
[media_player.spotify docs]: https://home-assistant.io/components/media_player.spotify
[media_player.webostv docs]: https://home-assistant.io/components/media_player.webostv
[modbus docs]: https://home-assistant.io/components/modbus
[mqtt docs]: https://home-assistant.io/components/mqtt
[mysensors docs]: https://home-assistant.io/components/mysensors
[neato docs]: https://home-assistant.io/components/neato
[notify.kodi docs]: https://home-assistant.io/components/notify.kodi
[notify.sendgrid docs]: https://home-assistant.io/components/notify.sendgrid
[notify.webostv docs]: https://home-assistant.io/components/notify.webostv
[scene.lifx_cloud docs]: https://home-assistant.io/components/scene.lifx_cloud
[sensor.cpuspeed docs]: https://home-assistant.io/components/sensor.cpuspeed
[sensor.crimereports docs]: https://home-assistant.io/components/sensor.crimereports
[sensor.fedex docs]: https://home-assistant.io/components/sensor.fedex
[sensor.fido docs]: https://home-assistant.io/components/sensor.fido
[sensor.modbus docs]: https://home-assistant.io/components/sensor.modbus
[sensor.mvglive docs]: https://home-assistant.io/components/sensor.mvglive
[sensor.neato docs]: https://home-assistant.io/components/sensor.neato
[sensor.opensky docs]: https://home-assistant.io/components/sensor.opensky
[sensor.qnap docs]: https://home-assistant.io/components/sensor.qnap
[sensor.speedtest docs]: https://home-assistant.io/components/sensor.speedtest
[sensor.systemmonitor docs]: https://home-assistant.io/components/sensor.systemmonitor
[sensor.uber docs]: https://home-assistant.io/components/sensor.uber
[sensor.ups docs]: https://home-assistant.io/components/sensor.ups
[sensor.usps docs]: https://home-assistant.io/components/sensor.usps
[sensor.vera docs]: https://home-assistant.io/components/sensor.vera
[sun docs]: https://home-assistant.io/components/sun
[switch.tplink docs]: https://home-assistant.io/components/switch.tplink
[switch.wake_on_lan docs]: https://home-assistant.io/components/switch.wake_on_lan
[switch.wemo docs]: https://home-assistant.io/components/switch.wemo
[telegram_bot docs]: https://home-assistant.io/components/telegram_bot
[telegram_bot.polling docs]: https://home-assistant.io/components/telegram_bot.polling
[telegram_bot.webhooks docs]: https://home-assistant.io/components/telegram_bot.webhooks
[tts.google docs]: https://home-assistant.io/components/tts.google
[tts.marytts docs]: https://home-assistant.io/components/tts.marytts
[vera docs]: https://home-assistant.io/components/vera
[wemo docs]: https://home-assistant.io/components/wemo
[zwave docs]: https://home-assistant.io/components/zwave
[zwave.node_entity docs]: https://home-assistant.io/components/zwave.node_entity

JesseWebDotCom committed:
Exposed more attributes, enabled play_media tv show or season episodes

JesseWebDotCom committed:
mend

Exposed more attributes, enabled play_media tv show or season episodes

JesseWebDotCom committed:
Fixed setting username and content rating data

JesseWebDotCom committed:
Fixed log errors if session username or content rating is blank

JesseWebDotCom committed:
String formatting, import order, type check fixes

emlove committed:
Update kodi for aiohttp2 (#6967)

aufano committed:
Fix current_temperature is rounded (#6960)

* Fix current_temperature is rounded

* fix  Unnecessary parens after 'if'

JesseWebDotCom committed:
import order fix

pvizeli committed:
Bugfix time and task coro (#6968)

* Bugfix time and task coro

* fix also other create_task

* fix tests

* fix lint in test

pvizeli committed:
Initial import for HassIO (#6935)

* Initial import for HassIO

* Cleanup api code for views

* First unittest for view

* Add test for edit view

* Finish unittest

* fix addons test

* cleanup service.yaml

* Address first round with ping command

* handle timeout dynamic

* fix lint

andrey-git committed:
Preserve customize glob order. (#6963)

* Preserve customize glob order.

* add tests

viswa-swami committed:
Foscam Camera: Adding exception handling when fetching the camera image to avoid python exception errors when host is not reachable or rather any url error to camera (#6964)

* Adding exception handling when fetching the camera image to avoid python errors when host is not reachable or any url errors to camera

* Added exception as ConnectionError instead of plain except

* Added exception as ConnectionError instead of plain except. Removed the unused error handle

rytilahti committed:
light.yeelight: catch i/o related exceptions from the backend lib (#6952)

Fixes/mitigates problems with #5949 and #6624

happyleavesaoc committed:
Crime Reports sensor (#6966)

* add crimereports

* add crimereports metadata

* implicit interval

* remove zone support

BillyNate committed:
From Dusk till Dawn (#6857)

* Added dawn, dusk, noon and midnight to the Sun component

* Created a helper method for the solar events

emlove committed:
Tests for zwave services (#6937)

* Initial tests for zwave services

* Fix linter issues

* Complete zwave service tests

balloob committed:
Fix control+c quitting HASS (#6974)

mezz64 committed:
Update Emby for aiohttp v2 (#6981)

rytilahti committed:
switch.tplink: upgrade to the newest upstream release which adds support for plugs using the newer communication protocol (#6979)

rytilahti committed:
switch.tplink: bump pyhs100 version requirement (#6986)

emlove committed:
Add tests for ZWaveDeviceEntityValues helper (#6978)

* Add tests for ZWaveDeviceEntityValues helper

* Add remaining coverage

tchellomello committed:
Bump Amcrest module to 1.1.8 (#6990)

Fixed traceback when calculating SD card percent storage

   self._state = self._camera.percent(sd_used[0], sd_total[0])
AttributeError: 'Http' object has no attribute 'percent'

happyleavesaoc committed:
update gstreamer (#6987)

balloob committed:
Warn if start takes a long time. (#6975)

* Warn if start takes a long time.

* ps - cleanup

* Tweak message

* Add tests

* Tweak messagE

balloob committed:
Upgrade to aiohttp 2.0.6 (#6992)

balloob committed:
Make discovery not block start (#6991)

* Make discovery not block start

* Fix tests

balloob committed:
Downgrade aiohttp to 205 (#6994)

pavoni committed:
Bump pywemo version. (#7004)

balloob committed:
Fix Synology camera content type (#7010)

balloob committed:
Fix two more instances of JSON parsing synology (#7014)

Add an optional extended description…

Commit:
Bump pyalarmdotcom to support new version of aiohttp (#7021)

Add an optional extended description…

fabaff committed:
Fix US states check (fixes #7015) (#7017)

balloob committed:
Remove deprecated remote classes (#7011)

* Remove deprecated remote classes

* Lint

* Fix tests

* Lint

fabaff committed:
Replace 'vendor_id' with 'arch' (fixes #7003) (#7023)

micw committed:
more tests for slugify (#7027)

emlove committed:
Additional ZWave coverage (#7024)

* Additional ZWave coverage

* setup_platform tests

happyleavesaoc committed:
bump ups version (#7033)

happyleavesaoc committed:
update usps version (#7035)

happyleavesaoc committed:
update fedex (#7034)

pezinek committed:
Google TTS can't read percent sign (#6971) (#7030)

micw committed:
Feature/min max improvements (#6786)

* Fix #6783, remove a test that makes no sense anymore

* Fix #6784

* Fix typo in docstring

* Fix handling of known->unknown state, extended test, fix lint errors

* Better handling of mismatch in unit of measurement.

Set state to "unkown" and unit of measurement to "ERR" if unit of measurement differs between aggregatet states.
Add entity_id to logged warning.

* Make icon configurable

* Fix typo

* Fix lint

* Fix lint

* Fix lint

* Add option to set entity_id on min_max sensor

* Fix lint logging-not-lazy

* Revert "Add option to set entity_id on min_max sensor"

This reverts commit 4685f266477dfaba92cf8a256ecfe62c929c877b.

* Revert "Make icon configurable"

This reverts commit fe45aec82d9a07e730bebb4d47dc9618fb695fe6.

* Fixes

fabaff committed:
Upgrade psutil to 5.2.2 (#7037)

fabaff committed:
Upgrade sendgrid to 4.0.0 (#7038)

DavidMStraub committed:
Missing line name restriction added (fixes #7039) (#7040)

amelchio committed:
Plug file leak on LIFX unregister (#7031)

* Plug file leak on LIFX unregister

The aiolifx 0.4.4 release closes its socket when the unregister callback is
called. This plugs a file descriptor leak but also means that we must be
careful to not use the device after it goes unavailable.

Also, when a light reappears, it has a new device that must be used.

* Do not test self.available in service calls

The core will learn to handle that.

pvizeli committed:
Make core to look avilable state of device on servicecall (#7045)

fabaff committed:
Remove configuration sample (#7048)

pvizeli committed:
Bugfix wait on start event (#7013)

* Bugfix wait on start event

* Paulus sugestion

* Change handling with stop_track_task

* Add new unittests

* Update test_core.py

pvizeli committed:
Bugfix slider (#7047)

* Bugfix slider

* Update input_slider.py

* Update input_slider.py

Commit:
Add MaryTTS platform (#6988)

* Add MaryTTS platform

* Fix lint error

* Doc link, config and formatting fixes

* Remove stuff not needed with aiohttp2

* Get rid of unnecessary else statement

MartinHjelmare committed:
Fix mysensors callback (#7057)

* Fix mysensors callback

* All messages was not triggering proper updates. Fix by checking all
  child value types each update.

* Upgrade mysensors dep

* Fix pickle persistence when upgrading.

balloob committed:
Constrain chardet to 2.3 (#7063)

micw committed:
Fix/slugify with german umlaut ss (#7029)

* more tests for slugify

* Fix german umlauts in slugify

* Update __init__.py

balloob committed:
Speed up aiohttp (#7064)

balloob committed:
Upgrade netdisco to 1.0.0rc2 (#7008)

* Upgrade netdisco to 1.0.0rc2

* fix tests

sander76 committed:
Telegram bot component (incl. webhook and polling platform) (#6913)

* first commit.

* removed pointless string statement

* manually removed  # homeassistant.components.telegram_webhooks from requirements_all.txt

* deleted obsolete file.

* coveragerc abc

fabaff committed:
Upgrade paho-mqtt to 1.2.2 (#7066)

pvizeli committed:
Fix handling with register callbacks on added_to_hass (#7067)

pvizeli committed:
Lutron. Bugfix callback registration. (#7042)

* Lutron. Bugfix callback registration.

* Change handling to event

hawk259 committed:
Adding AlarmDecoder platform (#6900)

* Added AlarmDecoder platform

* remove try/catch for generic execption

* Changes for @pvizeli, thanks for the review!

Removed _ prefix from normal function variables
Removed _hass as it will be set via .hass for us
Broke out the three config (socket, serial, usb) and use vol.Any
Added support for USB I think, don't have device, but should work
Removed components dictionary, was form old group all code that didn't work

* Fix hass string handling

turbokongen committed:
Add communication data attributes to Zwave node_entity (#6813)

* Add quality attribute to node

* Move quality to node_entity

* adjustments

* Line lenght

* flake8

* Cleanup and add each entry from getNodeStatistics as attribute

* resolve conflict

* Move NETWORK to hass.data

* Test Done

* Cleanup from review

* Resolve conflicts

* Review changes

* Long lines....

* blank line

* import-error-disable

* Update tests Part1... Again

* Hound

* Argh!

* Argh!

* YABBADABBADOOOOOOO!

* Enhance tests

* hound

* Resolve

* resolve tests...

andrey-git committed:
Add product_name attribute to zwave nodes. (#7071)

robbiet480 committed:
Merge pull request #6869 from JesseWebDotCom/dev

Exposed more attributes, enabled play_media tv show or season episodes

robbiet480 committed:
Bump braviarc version to 0.3.7 (#7078)

titilambert committed:
Fix account balance in fido sensor (#7077)

MrMep committed:
MQTT: Managing binary payloads (#6976)

* Managing binary payloads

Hello,
background: I wrote a HA camera component that gets the image from a binary payload. I'm testing it with Zanzito (https://play.google.com/store/apps/details?id=it.barbaro.zanzito) and it works apparently well: it gets the image and correctly displays it in the front-end.
But I had to make the changes I'm proposing here: the message was being blocked because the utf-8 decoding failed.
As far as I know, the utf-8 encoding is required for the topic, not for the payload. What I did here was try the utf-8 decoding, but even if unsuccessful, it dispatches the message anyway.
Is there anything else I'm missing?
thanks
Gianluca

* Update __init__.py

* Update __init__.py

* Update __init__.py

* git test - ignore

* Should work

* minor fixes

* updated mqtt/services.yaml

* added two tests, modified threaded subscribe

* removing polymer

* requested changes

* requested changes - minor fix

* security wrap around payload_file_path

* services.yaml updated

* removed file publishing

* minor fix

colinodell committed:
Bump qnapstats library version to 0.2.4 (#7085)

turbokongen committed:
Add support fo map data from Neato (#6939)

* Responsiveness

* Delay was not needed as commands does not return until done.

* Add support for cleaning maps and cleaning data

* Hound

* Docstring

* Update requirements

* Review changes

* External lib now returns the raw data.

* debug

* Sensor did not refresh

* Error handling

* Issue warning on connection error

* Update requirements

* Review changes

Commit:
Added initial support for IKEA Tradfri Gateway (#7074)

* Added initial support for IKEA Tradfri Gateway

* Pinned requirement

* Fixed lint-errors

* Added file to .coveragerc

* Trying to fix commit

* Fixed requirements_all again

* Minor reorder of code

* Minor reorder of code

* Made the changes suggested by @balloob

* Made the changes suggested by @balloob and removed debug

* Update tradfri.py

andrey-git committed:
Better thread safety in zwave node_entity (#7087)

* Better thread safety

* Update node_entity.py

robbiet480 committed:
Break product name up into manufacturer name and product name

robbiet480 committed:
Merge pull request #7094 from robbiet480/zwave-manufacturer-product-names

Break Z-Wave product name up into manufacturer name and product name

emlove committed:
Uber version bump (#7100)

emlove committed:
No product ids configured should fetch all ids (#7091)

robbiet480 committed:
Allow extra keys in zone config

fabaff committed:
Upgrade aiohttp to 2.0.7 (#7106)

fabaff committed:
Update file header, add const for defaults, and update log messages (#7110)

fabaff committed:
Upgrade speedtest-cli to 1.0.4 (#7105)

fabaff committed:
Remove globally disabled pylint issue and update docstrings (#7111)

robbiet480 committed:
Merge pull request #7109 from home-assistant/allow-extra-in-zone-config

Allow extra keys in zone config

emlove committed:
Use third-party lib aioautomatic for automatic (#7126)

fabaff committed:
Upgrade chardet to 3.0.2 (#7112)

amelchio committed:
Send stderr of ping tracker to devnull (#7096)

When pinging an inaccessible device, OS errors like

    ping: sendto: No route to host

can occur. For a ping tracker this is not an error but rather a normal
situation. Thus, it makes sense to hide the error.

tchellomello committed:
Upgraded Amcrest module to 1.1.9 to support new firmware versions: (#7130)

- V2.400.AC01.15.R.20170328
    - V2.420.AC00.17.R.20170322

MrMep committed:
Mqtt camera (#7092)

* first commit

* minor fixes

* minor fix

* async_camera_image + test

* minor fix

* async calls

MartinHjelmare committed:
Bump version to 0.43.0.dev0 (#7132)

balloob committed:
Load zwave panel (#7127)

* Load Z-Wave panel when component loads

* Update frontend

* Fix tests

Commit:
IKEA Tradfri Gateway: added support for RGB (#7115)

* After rebase and all fixes

* Added color_rgb_to_hex to util.color

* Added test_color_rgb_to_hex

* Changed reference to color_rgb_to_hex

* Bumped to pytradfri 0.5, having support for retry

* Bumped to pytradfri 0.5, having support for retry

* Bumped to pytradfri 0.5, having support for retry

* Bumped to pytradfri 0.5, having support for retry

* Rolled back to 0.4

* Rolled back to 0.4

balloob committed:
Make Tradfri discoverable (#7128)

* Make Tradfri discoverable

* Fix lint errors

* Fix bugs and clean up calls to light_control

* Add more color util tests

* Add coap client to dockerfile

robbiet480 committed:
Make version number optional and a string to fix identify issue introduced in iOS 1.0.1 (#7141)

emlove committed:
Fix for zwave RGB setting (#7137)

robbiet480 committed:
Replace rollershutter with cover in demo (#7140)

emlove committed:
Add debug logging to automation initial state (#7068)

amelchio committed:
Always return True/False from is_state and is_state_attr (#7138)

* Always return True/False from is_state and is_state_attr

These functions are documented to always return True/False but the
short-circuit evaluation would return None if the entity_id did not exist.

* Reword into a single statement

amelchio committed:
Add LIFX Cloud scene support (#7124)

This uses the LIFX HTTP API to list and activate the scenes that are
stored in the LIFX cloud by the native LIFX smartphone apps.

amelchio committed:
Fix LIFX lights with disappearing names (#7119)

* Cache the name of LIFX lights

After #7031 the LIFX device will change during an unregister/register
transition. This has the user-visible effect of the new device missing
a friendly name until the next poll.

We now cache the name internally and it will then transfer to the new
device when it registers.

* Allow LIFX logging even without an available device

This will allow us to set the device to None when it unregisters.

* Calculate LIFX availability from the existence of a device

This has become possible because the device is no longer needed
to provide the name of the light when it is unavailable.

We just have to forget the device when it unregisters.

thecynic committed:
lutron: fix typo that prevented callback registration (#7148)

pvizeli committed:
Fix HassIO timeout bug (#7155)

* Fix HassIO timeout bug

* fix lint

* Add long polling timeout to stop event

robbiet480 committed:
Fix iOS icon calculation to return mdi:battery for levels above 95

robbiet480 committed:
Change iOS sensor unique ID to use the device ID

robbiet480 committed:
Track device last identify time

Mister-Espria committed:
small fix for random effect in order to use the whole rgb range. So 255 is not excluded anymore. (#7156)

cmsimike committed:
exposed content_type in rest_command (#7101)

* exposed content_type in rest_command, which allows for manually specifying the content_type for more-strict api endpoints

* fixed up column length

Length was 86 chars, and it needed to be 79

* double import of HTTP_HEADER_CONTENT_TYPE

Removed the accidental double-import of HTTP_HEADER_CONTENT_TYPE

* moved rest_command-specific config value into component

* if no content_type, default to None

* unit test

* newline

* unused CONTENT_TYPE_TEXT_PLAIN

* removed the http-agnostic abstraction hass provided in favor of aiohttps hdrs constant

pavoni committed:
Add vera power meter. (#7134)

* Add vera power meter.

* Use W for power.

MartinHjelmare committed:
Add support for tradfri color temp (#7153)

* Add support for tradfri color temp

* Only return rgb_color if supported features indicate support.
* Return color_temp if supported features indicate support and dict
  with allowed color temperatures exist for manufacturer.
* Add manufacturer specific supported features, with default.
* Color temperature is set from a dict of allowed pairs per
  manufacturer, where kelvin is mapped to hex_color. When a user sets a
  color temperature, the closest matching temperature is selected and
  set.

* Clean up

balloob committed:
Disable MQTT camera test (#7164)

fabaff committed:
myStrom WiFi bulbs (#7161)

* Add initial support for myStrom WiFi bulbs

* Upgrade python-mystrom to 0.3.8

* Add myStrom light

* Fix lint issue

titilambert committed:
Fix #7026 adding a new wol parameter (#7144)

pavoni committed:
Add subscription update for Wemo switches, fix bug in Insight switches, fix wemo motion bug, fix wemo discovery (#7135)

* Fix wemo discovery.

* Bump wemo version, add subscription_update for basic switch, fix bug with turning insight switches off.

* Fix missed callback change for wemo motion.

* Regress netdisco changes.

michaelarnauts committed:
Update neato.py (#7166)

Fix leftover copy/paste error in comment of neato.py

molobrakos committed:
Keep track of already added players (#7149)

gurumitts committed:
Fix id zone mismatch (#7165)

* Fixes issue with id mismatch when multiple devices are connected to the lutron bridge

* Updates labels

* removes no longer needed config values.

* removes no longer needed imports

balloob committed:
Upgrade pytradfri to 1.0 (#7163)

aequitas committed:
Value of 0 should not be considered `unknown`. (#7139)

* Value of 0 should not be considered `unknown`.

* Reflect disconnect state in entity states.

* Due to adding unknown state on disconnect, the amount of reconnects can sometimes be more. Test for at least 2 reconnect attempts.

balloob committed:
Upgrade netdisco (#7171)

alexmogavero committed:
Added new services to platform kodi  (#6426)

* added new service

* fixed basic test in kodi platform

* Added new method async_get_albums

* Added new methods in module kodi

* added method find_song in kodi module

* method add_song_to_playlist made
async

* Added media type to method async_play_media

* added methods async_clear_playlist
and play_song

* methods play_song and find_song
made async

* added new service play_song

* Improved kodi._find
now it find for whole words only

* added possibility to specify artist in
kodi.async_find_artist

* added kodi.async_find_album

* added new optional input to play_song service

* In async_play_song added handling of no song found

* default artist value changed to ''

* async_add_song_to_playlist now can also
search for musinc

* added service add_song_to_playlist

* Added new service add_album_to_playlist

* added services to switch shuffle mode

* added service add_all_albums_to_playlist

* handled error in async_unset_shuffle
and async_set_shuffle

* Added abstract methods to media_player

* _server substituted with server property

* style made consistent with requirements

* Fixed issue with pylint

* Services moved to kodi platform

* service play_song removed

* removed service unset_shuffle

* all add services merged into one

* removed service get_artists

* added kodi_ to service names

* Fixed some style issues

* Removed changes in media_player __init__

* Implemented requested changes

* Fixed pylint problem

fabaff committed:
Supress trackback and upgrade PyMata to 2.14 (#7176)

fabaff committed:
Disable invalid-sequence-index (#7177)

fabaff committed:
Upgrade mypy to 0.501 (was renamed from mypy-lang) (#7117)

sytone committed:
Add condition for API failure (#7181)

* Add condition for API failure

If you are not running the latest ve3rsion of ZM this will cause exceptions to fire. This fix handles a response from ZM but a non successful attempt. 

This resolves the issue https://github.com/home-assistant/home-assistant/issues/7178

* Fixed houndci-bot issues

balloob committed:
Add history to component priority list (#7173)

balloob committed:
Tweak Tradfri (#7172)

balloob committed:
Update frontend

MrMep committed:
Mqtt camera test (#7175)

* mock mqtt

* minor fix

* minor fix

* minor fix

hmn committed:
updated pylgtv module to fix problems with timeouts (#7184)

balloob committed:
Fix wemo discovery (#7183)

* Fix wemo discovery

* Fix key

CharlesBlonde committed:
Add Bose soundtouch discovery support and upgrade libsoundtouch library (#7005)

* Add Bose soundtouch discovery support and upgrade libsoundtouch library

* Remove DEVICE global variable

* Update netdisco to lastest version

balloob committed:
Update frontend

happyleavesaoc committed:
spotify media player (#6980)

* spotify media player

* fix refresh token

* spotify improvements

* add checks for idle

* import STATE_IDLE

* support more media_ids, limit updates

* move spotify device update

* Remove schedule_update_ha_state because should_poll is true

MrMep committed:
JSON MQTT Device tracker (#7055)

* ready for PR

* minor fix

* another minor fix

* new platform mqtt_json

* using ATTR constants

* voluptuous check on JSON payload

* voluptuous check on JSON payload

happyleavesaoc committed:
opensky sensor (#7061)

* opensky sensor

* address opensky review comments

* update opensky distance calc

fabaff committed:
Add ping binary sensor (#7052)

* Add ping binary sensor

* Fix typo and lint issues

* Use SCAN_INTERVAL

soldag committed:
Added light.pwm component. (#7009)

* Added light.pwm component.

* Renamed pwm platform to rpi_gpio_pwm.

* Update requirements_all.txt

masarliev committed:
Fix for errors on missing preview on LG webos TV (#6755)

* fix for missing image preview on LG webos TV

* fix to use largeIcon if it start with http

postlund committed:
Fix auto discovery for Apple TV (#7188)

postlund committed:
Do not request artwork if not available (#7189)

This should fix the "dancing media player" issue where the media player
requests artwork when it's not really available, making the UI "dance".

mountainsandcode committed:
mvglive bug fixes and improvements (#6953)

* Refactored mvglive.py

This pull requests builds on the first work with the mvglive sensor:
- Refactoring the code so that multiple sensors for departures can be added
- Rewrites the transport mode restrictions ("products") to be more modular
- Fixes bugs, such as missing implementation of line restriction
- Other improvements, such as including data attribution

* Further improvements to mvglive sensor

- The API returns the property 'direction', which can be used to filter U-Bahn trains by direction without having to enter all final destinations
- The sensor icon now corresponds to the mode of transport of the next departure

* UBahnDirection refactored

U-Bahn SEV (bus replacement services) have unexpected direction values, fixed resulting bug and hound issues

fabaff committed:
Upgrade py-cpuinfo to 3.2.0 (#7190)

LvivEchoes committed:
Add support of input registers while querying modbus sensor. (#7082)

* Add support of input registers while querying modbus sensor.

* Changed config option. Refactoring.

pvizeli committed:
Add HassIO to discovery component (#7195)

* Add HassIO to autodiscovery

* Fix tests

* fix tests

* fix test v2

* fix mock test

* call

balloob committed:
Merge branch 'master' into dev

amelchio committed:
LIFX light effects (#7145)

* Refactor into find_hsbk

This will be useful for new methods that also have to find passed in colors.

* Add AwaitAioLIFX

This encapsulates the callback and Event that aiolifx needs and thus avoids an
explosion of those when new calls are added.

The refresh_state is now generally useful, so move it into its own method.

* Initial effects support for LIFX

These effects are useful as notifications. They mimic the breathe and pulse
effects from the LIFX HTTP API:

    https://api.developer.lifx.com/docs/breathe-effect
    https://api.developer.lifx.com/docs/pulse-effect

However, this implementation runs locally with the LIFX LAN protocol.

* Saturate LIFX no color value

Now the color is "full saturation, no brightness". This avoids a lot of
temporary white when fading from the "no color" value and into a real color.

* Organize LIFX effects in classes

This is to move the setup/restore away from the actual effect, making it quite
simple to add additional effects.

* Stop running LIFX effects on conflicting service calls

Turning the light on/off or starting a new effect will now stop the running
effect.

* Present default LIFX effects as light.turn_on effects

This makes the effects (with default parameters) easily accessible from
the UI.

* Add LIFX colorloop effect

This cycles the HSV colors, so that is added as an internal way to set a
color.

* Move lifx to its own package and split effects into a separate file

* Always show LIFX light name in logs

The name is actually the easiest way to identify a bulb so just using it
as a fallback was a bit odd.

* Compact effect getter

* Always use full brightness for random flash color

This is a stopgap. When a bit more infrastructure is in place, the intention
is to turn the current hue some degrees. This will guarantee a flash color
that is both unlike the current color and unlike white.

* Clear effects concurrently

We have to wait for the bulbs, so let us wait for all of them at once.

* Add lifx_effect_stop

The colorloop effect is most impressive if run on many lights. Testing
this has revealed the need for an easy way to stop effects on all lights
and return to the initial state of each bulb. This new call does just that.

Calling turn_on/turn_off could also stop the effect but that would not
restore the initial state.

* Always calculate the initial effect color

To fade nicely from power off, the breathe effect needs to keep an
unchanging hue. So give up on using a static start color and just find the
correct hue from the target color.

The colorloop effect can start from anything but we use a random color
just to keep things a little interesting during power on.

* Fix lint

* Update .coveragerc

pvizeli committed:
HassIO API v2 (#7201)

Add an optional extended description…

MrMep commented:
Yeee, thanks!
@balloob, isn't `JSON MQTT Device tracker` a new platform?

gl

balloob commented:
You're right it is. It didn't get the label initially because the initial version of the PR was not a new platform. Corrected it.

robbiet480 commented:
Didn't Crime Reports get added in last version? It's on the second line of new features: `Crime Reports sensor (@happyleavesaoc - #6966) (new-platform)`

amelchio committed:
Support xy_color with LIFX lights (#7208)

biacz committed:
Update ios.py (#7160)

* Update ios.py

as discussed. the part: 
       if battery_state == ios.ATTR_BATTERY_STATE_FULL:
            returning_icon_level = DEFAULT_ICON_LEVEL
kinda screws up the charging icon.

i might just miss a logical solution for that though.
let me know what you think. it might not be beautiful but i think its an overall improve over the current "double battery" solution

* Update ios.py

chound fix and full_battery_charge fix

* Update ios.py

removed new line

* Update ios.py

* Update ios.py

* Update ios.py

* Update ios.py

* Update ios.py

* Update ios.py

* merged request from robbie

* Update ios.py

* Update ios.py

* Update ios.py

sdague committed:
Fix arwn platform to update hass state when events are received (#7202)

The arwn platform was refactored to be asyncio friendly, however in
doing so one thing was missed which was explicitly telling hass when
something interesting has happened. This led to the very interesting
to debug issue that the state cards were all out of date, even though
the graphs were not.

hmn committed:
Issue 6749 updated pylgtv to 0.1.6 to fix thread leak in asyncio loop (#7199)

* updated pylgtv module to fix problems with timeouts

* - update pylgtv to 0.1.6
- handle new TimeoutError exception from pylgtv

* used full name for exception handling of concurrent.futures._base.TimeoutError

* the exception handling should now follow the rules

* float typecasting should not be necessary

* use asyncio for TimeoutError it’s an alias for concurrent.futures.TimeoutError

balloob committed:
tradfri: Improve color temp support detection (#7211)

MartinHjelmare committed:
Fix tradfri lights (#7212)

* Remove leftover use of slugify

* The IKEA manufacturer key is now exactly as found in device info.

* Fix bitwise addition of supported features

balloob committed:
Version bump to 0.43
